### DISCUSSION TITLE
V4 on Openwrt (arm7)

### DISCUSSION THREAD

**Discussion by roastduckkiller** [IrineSistiana/mosdns#450] [General]
V4 和前面的版本有一些不同的地方，下面是踩过的一些坑：
1. config.yaml 用 https://github.com/pmkol/easymosdns 项目里面的就可以了
2. 设备上面 /etc/mosdns 里面要建立一个目录 /rules，把所有的provider数据文件都放里面，确认上面的配置文件的路径和这个一致
3. 不要用 mosdns service install ，这个不work。要在/etc/init.d/目录建立文件 mosdns-init-Openwrt，内容如下（根据前面版本更改）： 

#!/bin/sh /etc/rc.common
# Copyright (C) 2020 IrineSistiana

START=18
USE_PROCD=1

#####  ONLY CHANGE THIS BLOCK  ######
PROG=/usr/bin/mosdns # where is mosdns
RES_DIR=/etc/mosdns/ # resource dir / working dir / the dir where you store ip/domain lists
CONF=./config.yaml   # where is the config file, it can be a relative path to $RES_DIR
#####  ONLY CHANGE THIS BLOCK  ######

start_service() {
  procd_open_instance
  procd_set_param command $PROG start --as-service -d $RES_DIR -c $CONF

  procd_set_param user root
  procd_set_param stdout 1
  procd_set_param stderr 1
  procd_set_param respawn "${respawn_threshold:-3600}" "${respawn_timeout:-5}" "${respawn_retry:-5}"
  procd_close_instance
  echo "mosdns is started!"
}

reload_service() {
  stop
  sleep 2s
  echo "mosdns is restarted!"
  start
}
4. 文件建立以后，运行 /etc/init.d/mosdns-init-Openwrt enable
5. 运行 /etc/init.d/mosdns-init-openwrt start
6. 也可以重启试试能不能自动运行service （ps 看看进程在不在）

**Comment by liangxyz** [comment]
mark,不知道直接在op上使用easymosdns可以不
